DSharpPlus/Net/Rest/RateLimitStrategy.cs

```diff
@@ -1,35 +1,37 @@
 using System;
 using System.Collections.Concurrent;
+using System.Collections.Generic;
 using System.Linq;
 using System.Net;
 using System.Net.Http;
 using System.Net.Http.Headers;
 using System.Threading;
 using System.Threading.Tasks;
 
-using Microsoft.Extensions.Caching.Memory;
 using Microsoft.Extensions.Logging;
 
 using Polly;
 
 namespace DSharpPlus.Net;
 
-using System.Collections.Generic;
-
-internal class RateLimitStrategy(ILogger logger, int waitingForHashMilliseconds = 200) : ResilienceStrategy<HttpResponseMessage>
+internal class RateLimitStrategy : ResilienceStrategy<HttpResponseMessage>
 {
     private readonly RateLimitBucket globalBucket = new(50, 50, DateTime.UtcNow.AddSeconds(1));
-    private readonly MemoryCache cache = new(new MemoryCacheOptions{ });
-    private readonly ConcurrentDictionary<string, string> routeHashes = new();
-    
-    /// <summary>
-    /// Collection of routes we are waiting for a hash. This is the case when we have a request for which we dont know the
-    /// hash but are waiting for a response to get it.
-    /// </summary>
-    private readonly List<string> waitingForHashRoutes = [];
-    private readonly SemaphoreSlim waitingForHashListSemaphore = new(1, 1);
-    
-    private static readonly TimeSpan second = TimeSpan.FromSeconds(1);
+    private readonly ConcurrentDictionary<string, RateLimitBucket> buckets = [];
+    private readonly ConcurrentDictionary<string, string> routeHashes = [];
+
+    private readonly ValueTask ratelimitCleanerTask;
+
+    private readonly ILogger logger;
+    private readonly int waitingForHashMilliseconds;
+
+    public RateLimitStrategy(ILogger logger, int waitingForHashMilliseconds = 200)
+    {
+        this.logger = logger;
+        this.waitingForHashMilliseconds = waitingForHashMilliseconds;
+
+        this.ratelimitCleanerTask = CleanAsync();
```

Comment:
Do we need to dispose this task somehow if this gets disposed

Reply:
i suppose we should, yes